flipt-io/flipt
# chore: update loadtesting to be more reflective of real world app

redoes our loadtest to do the following:

- vary the request payload for each call, allowing us to reflect real world usage better
- enable caching and auth separately to see how they affect performance
- change the rate of requests
- set log levels on both Flipt and Pyroscope to > DEBUG to remove extraneous output in the log
- run both variant eval (v1) and boolean eval (rollouts) separately for 60s each, outputting results at end

## Example:

```
$ FLIPT_AUTH_ENABLED="true" mage hack:loadTest

[EVALUATE]: Vegeta attack for 60.000000 seconds...
[EVALUATE]: Vegeta attack complete

Authentication Enabled: true
Cache Enabled: false
--------------------
Mean: 12.443488ms
95th percentile: 13.901631ms
99th percentile: 17.551379ms
Max: 45.307333ms
Requests: 6000
Throughput: 100.012008
Errors: []
Status Codes: map[200:6000]
Success: 100.000000 %


[BOOLEAN]: Vegeta attack for 60.000000 seconds...
[BOOLEAN]: Vegeta attack complete

Authentication Enabled: true
Cache Enabled: false
--------------------
Mean: 4.531645ms
95th percentile: 9.427763ms
99th percentile: 15.554324ms
Max: 32.3955ms
Requests: 6000
Throughput: 100.005388
Errors: []
Status Codes: map[200:6000]
Success: 100.000000 %
```
